tiledb/sm/filesystem/azure.cc

```diff
@@ -0,0 +1,716 @@
+  if (!request.valid()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("Flush blob failed on: " + uri.to_string())));
+  }
+
+  azure::storage_lite::storage_outcome<void> outcome = request.get();
+  if (!outcome.success()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("Flush blob failed on: " + uri.to_string())));
+  }
+
+  // Clear 'write_ss'.
+  std::stringstream().swap(*write_ss);
+
+  return Status::Ok();
+}
+
+Status Azure::is_empty_container(const URI& uri, bool* is_empty) const {
+  assert(client_);
+  assert(is_empty);
+
+  if (!uri.is_azure()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("URI is not an Azure URI: " + uri.to_string())));
+  }
+
+  std::string container_name;
+  RETURN_NOT_OK(parse_azure_uri(uri, &container_name, nullptr));
+
+  std::future<azure::storage_lite::storage_outcome<
+      azure::storage_lite::list_blobs_segmented_response>>
+      request = client_->list_blobs_segmented(container_name, "", "", "", 1);
+  if (!request.valid()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("List blobs failed on: " + uri.to_string())));
+  }
+
+  azure::storage_lite::storage_outcome<
+      azure::storage_lite::list_blobs_segmented_response>
+      outcome = request.get();
+  if (!outcome.success()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("List blobs failed on: " + uri.to_string())));
+  }
+
+  azure::storage_lite::list_blobs_segmented_response response =
+      outcome.response();
+
+  *is_empty = response.blobs.empty();
+
+  return Status::Ok();
+}
+
+Status Azure::is_container(const URI& uri, bool* const is_container) const {
+  assert(is_container);
+
+  if (!uri.is_azure()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("URI is not an Azure URI: " + uri.to_string())));
+  }
+
+  std::string container_name;
+  RETURN_NOT_OK(parse_azure_uri(uri, &container_name, nullptr));
+
+  return this->is_container(container_name, is_container);
+}
+
+Status Azure::is_container(
+    const std::string& container_name, bool* const is_container) const {
+  assert(client_);
+  assert(is_container);
+
+  std::future<azure::storage_lite::storage_outcome<
+      azure::storage_lite::container_property>>
+      request = client_->get_container_properties(container_name);
+  if (!request.valid()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("Get container properties failed on: " + container_name)));
+  }
+
+  azure::storage_lite::storage_outcome<azure::storage_lite::container_property>
+      outcome = request.get();
+  if (!outcome.success()) {
+    *is_container = false;
+    return Status::Ok();
+  }
+
+  azure::storage_lite::container_property response = outcome.response();
+
+  *is_container = response.valid();
+  return Status::Ok();
+}
+
+Status Azure::is_dir(const URI& uri, bool* const exists) const {
+  assert(client_);
+  assert(exists);
+
+  // Potentially add `/` to the end of `uri`
```

Comment:
I intended to match all semantics, so it would be an error if it was unmatched. In this scenario, it is the same. I copy+pasted all of the URI twiddling from s3.cc